benchmarks/benchmarks/memtrace_evaluation/__init__.py

```diff
@@ -0,0 +1,122 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the BSD license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Benchmarks to evaluate tracemalloc/rss methods of memory measurement."""
+
+from .. import TrackAddedMemoryAllocation
+from .memory_exercising_task import SampleParallelTask
+
+
+class MemcheckCommon:
+    # Basic controls over the test calculation
+    default_params = {
+        "measure": "tracemalloc",  # alternate: "rss"
+        "runtype": "threads",  # alternate: "processes"
+        "ysize": 10000,
+        "nx": 2000,
+        "nblocks": 6,
+        "nworkers": 3,
+    }
+
+    def _setup(self, **kwargs):
+        params = self.default_params.copy()
+        params.update(kwargs)
+        measure = params["measure"]
+        runtype = params["runtype"]
+        ysize = params["ysize"]
+        nx = params["nx"]
+        nblocks = params["nblocks"]
+        nworkers = params["nworkers"]
+
+        nyfull = ysize // nblocks
```

Comment:
It looks like you're applying this division twice, I think that's why adding blocks has such an exagerated effect on performance.

Reply:
Oh, yes I see !
Hopefully that may explain the peculiar behaviour of the timings too.

I will fix this and re-investigate ...

Reply:
https://github.com/SciTools/iris/pull/5946/commits/84ff4ad5d6cf9c8b95a204dd563764c36ec27f69 fixes, I think